### PR TITLE
Failing REFL014 test involving a member not visible to the analyzer

### PR DESCRIPTION
**Pull request by jnm2**
Re: https://github.com/DotNetAnalyzers/ReflectionAnalyzers/issues/189#issuecomment-439706200

### CODE REVIEW

**Comment by JohanLarsson** [comment]
Beautiful, thanks!

File: ReflectionAnalyzers.Tests/TestHelper.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            var binaryReferencedCompilation = CSharpCompilation.Create(
+                "BinaryReferencedAssembly",
+                new[] { CSharpSyntaxTree.ParseText(code) },
+                AnalyzerAssert.MetadataReferences,
+                new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+            using (var binaryReferencedContent = new MemoryStream())
+            {
+                var binaryEmitResult = binaryReferencedCompilation.Emit(binaryReferencedContent);
+                Assert.That(binaryEmitResult.Diagnostics, Is.Empty);
```
Comment (jnm2) [inline ReflectionAnalyzers.Tests/TestHelper.cs]: Should this be more lenient?

Comment (jnm2) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: Something you can figure out when it goers into the assert lib.

Comment (JohanLarsson) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: Oh, missed that, makes sense to check there I think. Should help finding test bugs.

Comment (JohanLarsson) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: Maybe you write the PR for the assert lib?

Comment (jnm2) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: Maaaybe...you'd have the best sense of how this fits into the big picture of that API, though.

Comment (jnm2) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: My preference these days is to not split into libs until the usage patterns have proven themselves in public projects.

Comment (jnm2) [reply ReflectionAnalyzers.Tests/TestHelper.cs]: Maaaybe...you'd have the best sense of how this fits into the big picture of that API, though.

My preference these days is to not split into libs until the usage patterns have proven themselves in multiple projects.

**Pull request merged**